lib/model/folder.go

```diff
@@ -238,3 +251,25 @@ func (f *folder) setError(err error) {
 
 	f.stateTracker.setError(err)
 }
+
+// Helper function to check whether either the ignorePerm flag has been
+// set on the local host or the FlagNoPermBits has been set on the file/dir
+// which is being pulled.
+func (f *folder) ignorePermissions(file protocol.FileInfo) bool {
+	return f.IgnorePerms || file.NoPermissions
```

Comment:
Micro-nit: I'm not sure this actually merits a helper function. The code is shorter and clearer than the comment.